sequencer/src/state_signature.rs

```diff
@@ -38,47 +42,73 @@ pub(super) async fn state_signature_loop<N>(
+                    Ok(state) => {
+                        let signature = context.sign_new_state(&state).await;
+                        tracing::debug!(
+                            "New leaves decided. Latest block height: {}",
+                            leaf.get_height(),
+                        );
 
-                if let Some(client) = client {
-                    let request_body = StateSignatureRequestBody {
-                        key: key.clone(),
-                        state,
-                        signature,
-                    };
-                    if let Err(error) = client
-                        .post::<()>("api/state")
-                        .body_binary(&request_body)
-                        .unwrap()
-                        .send()
-                        .await
-                    {
-                        tracing::warn!("Error posting signature to the relay server: {:?}", error);
+                        if let Some(client) = client {
+                            let request_body = StateSignatureRequestBody {
+                                key: key.clone(),
+                                state,
+                                signature,
+                            };
+                            if let Err(error) = client
+                                .post::<()>("api/state")
+                                .body_binary(&request_body)
+                                .unwrap()
+                                .send()
+                                .await
+                            {
+                                tracing::warn!(
+                                    "Error posting signature to the relay server: {:?}",
+                                    error
+                                );
+                            }
+                        }
                     }
+                    Err(err) => tracing::error!("Error generating light client state: {:?}", err),
                 }
             }
         }
     }
     tracing::info!("And now his watch has ended.");
 }
 
+fn hash_bytes_to_field(bytes: &[u8]) -> Result<CircuitField, PrimitivesError> {
+    // make sure that `mod_order` won't happen.
+    let bytes_len = ((<CircuitField as PrimeField>::MODULUS_BIT_SIZE + 7) / 8 - 1) as usize;
+    let elem = bytes
+        .chunks(bytes_len)
+        .map(CircuitField::from_le_bytes_mod_order)
+        .collect::<Vec<_>>();
+    Ok(VariableLengthRescueCRHF::<_, 1>::evaluate(elem)?[0])
+}
+
 fn form_light_client_state(
     leaf: &Leaf,
     stake_table_comm: &StakeTableCommitmentType,
-) -> LightClientState {
-    // TODO(Chengyu): fill these `default()` with actual value
-    LightClientState {
+) -> Result<LightClientState, PrimitivesError> {
+    let header = leaf.get_block_header();
+    let mut block_comm_root_bytes = vec![];
+    header
+        .block_merkle_tree_root
+        .serialize_compressed(&mut block_comm_root_bytes)?;
```

Comment:
why is the block header being the block commitment root?
we are supposed to build a merkle tree out of all the block commitment, right?

Reply:
Yes and this is the Merkle root. Just naming

Reply:
i think the BlockHeader is [more than just block_commit_root](https://github.com/EspressoSystems/espresso-sequencer/blob/main/sequencer/src/header.rs#L79), right? even though I don't see a problem, but it's not technically this bytes is committing to an superset of system states than just block_comm_root.